test/object.js

```diff
@@ -670,6 +670,23 @@ describe('object', function () {
                 done();
             });
         });
+
+        it('should ignore a key with ignoredUndefined if from does not exist', function(done){
+
+            var schema = Joi.object().rename('b', 'a', { ignoreUndefined: true });
+
+            var input = {
+                a: 'something'
+            };
+
+            schema.validate(input, function (err, value) {
+
+                expect(err).to.not.exist();
+                expect(Object.keys(value)).to.include('a');
+                expect(value.a).to.equal('something');
```

Comment:
That's right, I just copied/paste https://github.com/JbIPS/joi/blob/master/test/object.js#L652.
If you think deep.equal() is better, I'll change it